File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/_4_BastardInjection/InMessages/StopMessage.java
package com.github.grzesiek_galezowski.BastardInjection._4_BastardInjection.InMessages;

class StopMessage implements AcmeMessage {
  public void AuthorizeUsing(IAuthorization authorizationRules) {
    System.out.println("Authorizing Stop with " + authorizationRules);
  }

  public void WriteTo(DataDestination dataDestination) {
    System.out.println("Writing Stop to " + dataDestination);
  }
}

File: CSharp/DependencyInjection/DependencyInjection/BastardInjection/_4_BastardInjection/InMessages/StopMessage.cs
using System;
using BastardInjection._4_BastardInjection.Interfaces;
using BastardInjection._4_BastardInjection.Services;

namespace BastardInjection._4_BastardInjection.InMessages
{
  class StopMessage : AcmeMessage
  {
    public void AuthorizeUsing(IAuthorization authorizationRules)
    {
      Console.WriteLine("Authorizing Stop with " + authorizationRules);
    }

    public void WriteTo(DataDestination dataDestination)
    {
      Console.WriteLine("Writing Stop to " + dataDestination);
    }
  }
}